utility/mm2_emergency-expire-repo

```diff
@@ -0,0 +1,108 @@
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+"""
+This script is an emergency script meant for releng to be used if an very
+important update has went out, and we want to make sure that clients pick up
+the latest version of the repo, even if that means more load on the master
+mirrors.
+It does this by deleting all non-current file_detail entries for this specific
+repository.
+"""
+
+import logging
+import argparse
+import sys
+
+import mirrormanager2.lib
+from mirrormanager2.lib.model import Repository, Product, Version
+
+
+logger = logging.getLogger('mm2')
+
+
+def setup_logging(options):
+    format = "[%(asctime)s][%(name)10s %(levelname)7s] %(message)s"
+    datefmt = '%m/%d/%Y %I:%M:%S %p'
+    logging.basicConfig(format=format, datefmt=datefmt)
+
+    if options.debug:
+        logger.setLevel(logging.DEBUG)
+    else:
+        logger.setLevel(logging.INFO)
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Expire old files for repos")
+    parser.add_argument(
+        'product',
+        help="Product to clear old filedetails for (Fedora, EPEL)")
+    parser.add_argument(
+        'version',
+        help="VERSION to clear old filedetails for (20, 21)")
+    parser.add_argument(
+        "-c", "--config",
+        dest="config", default='/etc/mirrormanager/mirrormanager2.cfg',
+        help="Configuration file to use")
+    parser.add_argument(
+        "--debug",
+        dest="debug", default=False, action="store_true",
+        help="enable debugging")
+
+    options = parser.parse_args()
+
+    config = dict()
+    with open(options.config) as config_file:
+        exec(compile(config_file.read(), options.config, 'exec'), config)
+
+    session = mirrormanager2.lib.create_session(config['DB_URL'])
+
+    setup_logging(options)
+
+    product = session.query(Product).filter_by(name=options.product).first()
+    if not product:
+        logger.error('No product %s found' % options.product)
+        return 1
+
+    version = session.query(Version).filter_by(name=options.version,
+                                               product_id=product.id).first()
+    if not version:
+        logger.error('No version %s found' % options.version)
+        return 1
+
+    repos = session.query(Repository).filter_by(version_id=version.id).all()
+    if len(repos) < 1:
+        logger.error('No repos found for %s version %s' % (product, version))
+        return 1
+
+    for repo in repos:
+        logger.info("Clearing for repo %s" % repo.name)
+        (files, deleted) = repo.emergency_expire_old_file_details(session)
+        logger.info("Files kept: %s, deleted: %s" % (files, deleted))
```

Comment:
Is this going to be readable?

As in, if there are really a lot of files deleted, can the output be read in a meaningful way?